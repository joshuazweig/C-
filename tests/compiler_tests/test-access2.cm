int main()
{

	stone a;
	stone b;
	stone c;
	stone d;

	stone a1;
	stone b1;
	stone c1;
	stone d1;

	mint m;
	mint n;

	//curve cc;

	a = "3";
	b = "5";
	c = "37";
	d = "101";

	m = <a, b>;
	n = <c, d>;

	//cc = <m, n>;

	printf("Calling print_stone on a (=3): ");
	print_stone(a);

	printf("Calling print_stone on b (=5): ");
	print_stone(b);

	printf("Calling print_stone on c (=37): ");
	print_stone(c);

	printf("Calling print_stone on d (=101): ");
	print_stone(d);

	a1 = access_curve(<m, n>, 0);
	printf("Printing result of access_curve(cc, 0), should = a: ");
	print_stone(a1);

	b1 = access_curve(<m, n>, 1);
	printf("Printing result of access_curve(cc, 1), should = b: ");
	print_stone(b1);

	c1 = access_curve(<m, n>, 2);
	printf("Printing result of access_curve(cc, 2), should = c: ");
	print_stone(c1);

	d1 = access_curve(<m, n>, 3);
	printf("Printing result of access_curve(cc, 3), should = d: ");
	print_stone(d1);

		
	return 0;
}
