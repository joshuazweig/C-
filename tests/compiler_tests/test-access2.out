Calling print_stone on a (=3): 3
Calling print_stone on b (=5): 5
Calling print_stone on c (=37): 37
Calling print_stone on d (=101): 101
Printing result of access_curve(cc, 0), should = a: 3
Printing result of access_curve(cc, 1), should = b: 5
Printing result of access_curve(cc, 2), should = c: 37
Printing result of access_curve(cc, 3), should = d: 101